feat: add missing admin template blocks

JoshuaBehrens opened the pull request:
### 0. Preamble

This is a follow up on https://github.com/shopware/shopware/pull/3116 which has been declined for the reason of "we are not doing this anymore" on the other hand in the past year I still have reasons to need these and want to bring this up to the table again.

Reasons:
* Replace the HTML preview components to not use the editor
* Change the app error page to add a more suitable explanation for certain apps
* Change the sw-inactivity-login ( https://github.com/HEPTACOM/HeptacomShopwarePlatformAdminOpenAuth/commit/6c37b72e3010e5800a93f23b6620a569ab6fac0b )
* Add new custom entity input field types
* Change display of flow templates to highlight those, who are relevant for the project
* Disable the sw-settings-usage-data-general - it is not enough to disable admin, you also have to hide it from anywhere
* Change instructions shown on the license violation how to solve them
* Add fields to the newsletter form CMS element
* sw-extension-my-extensions account to explain, what login data has to be used there and that it is ok, to enter it there although it is not account.shopware.com
* sw-order-customer-address-select improve the customer address selection with support for new fields
* You already have about 7k blocks, I just want less than more

There are already others who are experiencing the same issue:

* NEXT-38504 - https://github.com/shopware/shopware/issues/4896 (which seems to be fixed already?)
* In the past we have had support for this way to code https://issues.shopware.com/issues/NEXT-13056
* Changes like this have been approved https://github.com/shopware/shopware/pull/3793

The docs state no alternative not event a concept with an ADR:

* We still have documentation like this:
    * https://github.com/shopware/docs/blob/359c3c03f351d25abbfa5af04fb11822fda6c441/guides/plugins/plugins/administration/add-new-tab.md?plain=1#L43-L77
    * https://github.com/shopware/docs/blob/359c3c03f351d25abbfa5af04fb11822fda6c441/guides/plugins/plugins/administration/search-custom-data.md?plain=1#L91-L113

In the past @jleifeld and @mstegmeyer have been looking into these topics. Maybe we can talk about it once again.

---

### 1. Why is this change necessary?

Although the twig blocks in the admin vue component are considered private there is a downside of the implementation of twigjs. There is always an extend of the component. So there is not way to change the template of a component, when there is no block. I looked up every component without a block and added at least a single block around it so you can have some sort of impact.

<img width="650" alt="image" src="https://github.com/shopware/platform/assets/1133593/ae1e0b31-bd41-4361-a844-a8094f72b082">


### 2. What does this change do, exactly?

Add twig blocks on every file, that has none at all.


### 3. Describe each step to reproduce the issue or behaviour.

* Create custom fields on units
* Open custom fields
* Look for the custom field set renderer showing the custom fields

<img width="638" alt="image" src="https://github.com/shopware/platform/assets/1133593/0e7e57ae-a87e-4ee0-a006-903bb407a818">

* Accept fate, that the custom field set renderer has not been placed
* No worries, it is easy in use. Just override the component with a new component

https://github.com/shopware/platform/blob/5faa66b6d7379bc5124681fa0fe16323d930cee8/src/Administration/Resources/app/administration/src/module/sw-settings-units/page/sw-settings-units-detail/index.html.twig#L1-L49

* hello_darkness_my_old_friend.wma

This is why I did this so I can do https://github.com/shopware/platform/pull/3117 easier


### 4. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

JoshuaBehrens committed:
Add missing admin template blocks

And rename one index.html.twig to sw-settings-units-detail.html.twig

mstegmeyer commented:
I think, the opinion of @seggewiss is also important on this, I was just involved in this, because it came into our area.

I have seen several added blocks here on private components, e.g. `sw-ai-copilot-badge`. Maybe could this be a differentiator? Only add blocks on non-private components?

JoshuaBehrens commented:
I do not want to undermine @seggewiss ! I was not sure he also working on this part. Great to have him in the discussion.

I understood the previous reason as "maintaining things, that change a lot is a hassle, so we rather don't have the entrypoints at all" So does it really care what you think is public or private? To me everything is public as it is readable in the vendor folder and not happy to be patched ^^' But on the other hand it would be better to have any entrypoint at all even if it means, that they are not covered by BC promise (unless it means breaking change promise :P ).

seggewiss commented:
@mstegmeyer @JoshuaBehrens 

Thank you for involving me in the conversation 😊 I have some great News to share! The public communication for this is still on the way but I can spoil it a little bit 😁 Blocks no longer are a problem as we found a native VueJs solution for the future.

**But**, adding blocks to private components is not a good idea. This suggests that these blocks are public API **but we can and will break those.** You just need remind yourself of this at all times!

JoshuaBehrens commented:
@seggewiss this is why I wrote it like that in the changelog. I do not care if you break them regularly as a project developer I don't need to maintain multiple versions at once :) Native Vue way is very nice. I am hyped to see this working with multiple extensions at the same component WOOP

seggewiss commented:
Hey @JoshuaBehrens 👋 

Thanks for your contribution! We'll get this PR scheduled for import. However, I wanted to give you and other developers a friendly heads-up:

The blocks in private components are nothing but a temporary solution, most of the blocks added in this PR are planned to be removed in version 6.9 without replacement. Feel free to use them for now if needed, but keep in mind they're not guaranteed to be available through to the 6.9 release.

Hope this helps with your current development needs! Let us know if you have any questions. 🙂

JoshuaBehrens commented:
@seggewiss glad to see this merged and thank you for the heads-up. So up until 6.9 we need to make sure to challenge meteor until we are happy?

jleifeld commented:
I will take care of the PR. Can you rebase the PR because there are some changes which would get otherwise reverted, e.g. 
<img width="1162" alt="Bildschirmfoto 2024-11-13 um 09 04 25" src="https://github.com/user-attachments/assets/497ecf4a-2325-4bf6-a1d6-df37879d20d7">

jleifeld reviewed (changes requested):
See https://github.com/shopware/shopware/pull/4942#issuecomment-2472752903

seggewiss commented:
@JoshuaBehrens Pinging you again to update the pull request 💙

seggewiss commented:
Hey @JoshuaBehrens, a quick head's up the last interaction is now older than 4 weeks from your side. If we don't hear from you until Friday 10th of January, I will close this pull request.

JoshuaBehrens commented:
@seggewiss @jleifeld I am sorry for the prolonged delay. I am happy about the upcoming changes in the next releases. Let's make sure we have full support for blocks in the last moments :)

I added skip-link as it was not covered as well, rebased and made sure the tests are all green :)
